### ISSUE TITLE
Ghost Parameter — `accumulation_steps` Accepted but Never Used

### ISSUE THREAD

**Issue by Afnan-abdi** [dataxid/dataxid-python#21]
## Description
`ModelConfig` exposes an `accumulation_steps: int = 1` parameter that
implies gradient accumulation support — a standard deep learning
technique that allows users to simulate large effective batch sizes on
memory-constrained hardware. A user setting
`ModelConfig(accumulation_steps=8, batch_size=32)` expects the training
to behave as if `batch_size=256`, making large-scale training feasible
on limited GPU/CPU memory.

A full codebase scan confirms that `accumulation_steps` is defined once
in `training/_config.py` and referenced in **zero other modules**. It
is never extracted in `Model.create()`, never passed to
`train_frozen()`, and never included in the `init-training` API
payload. The parameter is accepted without error, stored in the config
object, and silently discarded.

This is structurally identical to the previously reported ghost `seed`
parameter, but with more severe practical impact: `seed` affects
reproducibility, while `accumulation_steps` affects whether training
is *possible at all* on memory-constrained hardware. Users who cannot
fit large batch sizes in memory and set this parameter as a workaround
receive no benefit and no warning — the SDK offers a solution that does
not exist. Combined with the ghost `seed` parameter, this reveals a
systemic pattern: ModelConfig fields are never validated against actual
pipeline usage, meaning additional parameters may be silently
disconnected.

## Steps to Reproduce

1. Confirm the parameter exists in ModelConfig:
```python
   import dataclasses, dataxid
   fields = {f.name for f in dataclasses.fields(dataxid.ModelConfig)}
   print('accumulation_steps' in fields)  # True
```

2. Confirm it is never extracted in Model.create():
```python
   import inspect
   from dataxid.training import _model
   src = inspect.getsource(_model.Model.create)
   print('accumulation_steps' in src)  # False
```

3. Confirm it is never sent in the API payload:
```python
   from dataxid.training import _frozen
   src = inspect.getsource(_frozen.train_frozen)
   print('accumulation_steps' in src)  # False
```

4. Confirm it is absent from the entire codebase:
```python
   import pkgutil, importlib, inspect, dataxid
   found = [m for _, m, _ in pkgutil.walk_packages(
       dataxid.__path__, dataxid.__name__ + '.')
       if 'accumulation_steps' in inspect.getsource(
           importlib.import_module(m))]
   print(found)  # ['dataxid.training._config'] only
```

## Expected Behavior
`ModelConfig(accumulation_steps=N, batch_size=B)` simulates an
effective batch size of `N * B` by accumulating gradients across N
forward passes before each weight update. This is standard PyTorch
behavior and is the documented purpose of this parameter.

A user on a machine with 8GB RAM setting:
```python
config = dataxid.ModelConfig(
    batch_size=32,           # fits in memory
    accumulation_steps=8,    # simulates batch_size=256
)
```
should receive training equivalent to `batch_size=256`.

## Actual Behavior
```python
config = dataxid.ModelConfig(accumulation_steps=8)
print(config.accumulation_steps)  # 8 — accepted, stored
# But during training: parameter is silently discarded
# Effective batch size is always batch_size, never batch_size * N
```
The parameter is accepted without error or warning. Training proceeds
with the raw `batch_size` value only. No documentation states this
parameter is unimplemented.

## Fix Required
Three locations need updating:

**1. `training/_model.py` — extract the parameter:**
```python
accumulation_steps = config['accumulation_steps']
```

**2. `training/_frozen.py` — pass to train_frozen():**
```python
def train_frozen(
    model: Model,
    batch_size: int,
    max_epochs: int,
    early_stop_patience: int = 4,
    val_split: float = 0.1,
    accumulation_steps: int = 1,  # add this
) -> None:
```

**3. `training/_frozen.py` — include in init-training payload:**
```python
model._client.post(f"/v1/models/{model.id}/init-training", json={
    "max_epochs": max_epochs,
    "patience": early_stop_patience,
    "learning_rate": initial_lr,
    "label_smoothing": model._config["label_smoothing"],
    "embedding_dropout": model._config["embedding_dropout"],
    "time_limit_seconds": model._config["time_limit_seconds"],
    "accumulation_steps": accumulation_steps,  # add this
    "batches": batches,
})
```

**4. Systemic fix recommendation:**
Add a CI/CD validation test that asserts every field defined in
`ModelConfig` is referenced in at least one of `Model.create()` or
`train_frozen()`. This prevents future ghost parameters from being
introduced silently.

## Environment
- dataxid version: 0.2.0
- Python version: 3.12.13
- OS: Linux 6.6.113+ (Google Colab)
